kolibri/core/device/serializers.py

```diff
@@ -244,6 +247,23 @@ class Meta:
     def create(self, validated_data):
         raise serializers.ValidationError("Device settings can only be updated")
 
+    def update(self, instance, validated_data):
+        extra_settings = validated_data.get("extra_settings")
```

Comment:
Perhaps conditionalize on the existence of `extra_settings` just in case the serializer handles a partial update to the device settings where `extra_settings` wasn't modified.